packages/@aws-cdk/assertions/README.md

```diff
@@ -283,6 +321,37 @@ assert.hasResourceProperties('Foo::Bar', Match.objectLike({
 }});
 ```
 
+## Capturing Values
+
+This matcher APIs documented above allow capturing values in the matching entry
+(Resource, Output, Mapping, etc.). The following code captures a string from a
+matching resource.
+
+```ts
+// Given a template -
+// {
+//   "Resources": {
+//     "MyBar": {
+//       "Type": "Foo::Bar",
+//       "Properties": {
+//         "Fred": ["Flob", "Cat"],
+//         "Waldo": ["Qix", "Qux"],
+//       }
+//     }
+//   }
+// }
+
+const fredCapture = new Capture();
+const waldoCapture = new Capture();
+assert.hasResourceProperties('Foo::Bar', {
+  Fred: fredCapture,
+  Waldo: ["Qix", waldoCapture],
+});
+
+fredCapture.asArray(); // returns ["Flob", "Cat"]
+waldoCapture.asString(); // returns "Qux"
```

Comment:
I would appreciate a more "useful" example here that demonstrates the benefit of using capturing. We can use the assertion found in the pipelines module like `expect(captured.asString().length()).toBe(3)` or something